beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -59,6 +59,7 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 	randGen := rand.NewGenerator()
 	for _, slot := range slots {
 		ctx, span := trace.StartSpan(ctx, "processPendingBlocks.InnerLoop")
+		defer span.End()
```

Comment:
Won't this end up calling span.End() multiple times? The defer only happens when the function returns, not when the loop ends.

One way to fix this is to abstract the inner logic to its own function.

Reply:
That's what i was afraid of, thanks for confirming. I'll fix it in the draft

Reply:
As it turns out, we already took care of `span.End()` individually on `return` and `continue`. Although we did miss a few spots so I added them. Thanks